ext/featurefmt/rpm/contentmanifest/repositorycpe.go

```diff
@@ -0,0 +1,39 @@
+package contentmanifest
+
+import (
+	"github.com/quay/clair/v3/pkg/envutil"
+	log "github.com/sirupsen/logrus"
+)
+
+// RepoCPEUpdater provides interface for providing a mapping
+// between repositories and CPEs
+type RepoCPEUpdater interface {
+	Update() error
+	Get([]string) ([]string, error)
+}
+
+// RepoCPEMapping struct handles translation of repositories to CPEs
+type RepoCPEMapping struct {
+	Updater RepoCPEUpdater
+}
+
+var (
+	// MappingFileURL is URL to a mapping file with default value
+	MappingFileURL = envutil.GetEnv(
+		"REPO_CPE_MAPPING_URL",
+		"https://www.redhat.com/security/data/metrics/repository-to-cpe.json",
+	)
+)
+
+// RepositoryToCPE translates repositories into CPEs
+func (mapping *RepoCPEMapping) RepositoryToCPE(repositories []string) ([]string, error) {
+	log.WithField("repositories", repositories).Debug("Translating repositories into CPEs")
+	if len(repositories) == 0 {
+		return []string{}, nil
+	}
+	err := mapping.Updater.Update()
```

Comment:
If you embed the Updater (RepoCPEUpdater) this will read a little nicer.

```
mapping.Update()
```

Reply:
Fixed